test/e2e/testutils/net/svc.go

```diff
@@ -47,16 +47,17 @@ func EnsureNodePort(ctx context.Context, client kubernetes.Interface, clusterID,
 		Spec:   serviceSpec,
 		Status: v1.ServiceStatus{},
 	}
-	nodePort, err := client.CoreV1().Services(namespace).Create(ctx, nodePort, metav1.CreateOptions{})
+	_, err := client.CoreV1().Services(namespace).Create(ctx, nodePort, metav1.CreateOptions{})
 	if kerrors.IsAlreadyExists(err) {
-		nodePort, err = client.CoreV1().Services(namespace).Get(ctx, name, metav1.GetOptions{})
+		_, err = client.CoreV1().Services(namespace).Get(ctx, name, metav1.GetOptions{})
 		if err != nil {
 			klog.Error(err)
 			return nil, err
 		}
 	}
+	nodePort, err = client.CoreV1().Services(namespace).Get(ctx, name, metav1.GetOptions{})
```

Comment:
Curiosity. Why is this second get required?

Reply:
I don't know, @palexster do you?